### PR TITLE
Document params-file and improve self-update 

### PR DESCRIPTION
**Pull request by abhi18av**
Updated the documentation for `--params-file` and `self-update`

### CODE REVIEW

File: docs/cli.rst [hunk 1]
```diff
@@ -1284,7 +1290,8 @@ Update the nextflow runtime to the latest available version.
 
 **Description**
 
-The ``self-update`` command directs the ``nextflow`` cli to update itself to the latest stable release.
+The ``self-update`` command directs the ``nextflow`` cli to update itself to the latest stable release. 
+The ``self-update`` command also updates the `Nextflow launcher script <https://github.com/nextflow-io/nextflow/blob/master/nextflow>`_ in addition to the Java artifacts, whereas the ``NXF_VER`` environment variables only updates the Java artifacts.
```
Comment (abhi18av) [inline docs/cli.rst:1294]:
```suggestion
This command also updates the `Nextflow launcher script <https://github.com/nextflow-io/nextflow/blob/master/nextflow>`_ in addition to the Java artifacts, whereas the ``NXF_VER`` environment variables only updates the Java artifacts.
```

File: docs/cli.rst [hunk 2]
```diff
@@ -1284,7 +1290,8 @@ Update the nextflow runtime to the latest available version.
 
 **Description**
 
-The ``self-update`` command directs the ``nextflow`` cli to update itself to the latest stable release.
+The ``self-update`` command directs the ``nextflow`` cli to update itself to the latest stable release. 
+This command also updates the `Nextflow launcher script <https://github.com/nextflow-io/nextflow/blob/master/nextflow>`_ in addition to the Java artifacts, whereas the ``NXF_VER`` environment variables only updates the Java artifacts.
```
Comment (pditommaso) [inline docs/cli.rst:1294]: Don't think this is correct. The `self-update` is the same of `-self-update` and they do *not* update the launcher

Comment (abhi18av) [reply docs/cli.rst:1294]: Whoops!

@pditommaso , then let me cofirm my understanding here

1. `-self-update` updates only java artifacts
2. `NXF_VER` updates java artifacts and the launcher script

Therefore when `NXF_EDGE=1 nextflow -self-update`  is used, only the java artifacts are updated to the latest edge version.

Is this correct?

File: docs/cli.rst [hunk 3]
```diff
@@ -1265,6 +1265,12 @@ facilitates rapid iterations, inspections of any pipeline as well as debugging.
     $ nextflow run nextflow-io/hello -with-tower
  
 
+- Invoke the nextflow pipeline execution with a custom parameters ``YML`` file. 
+The parameters which are specified through this mechanism are merged with the resolved configuration (base configuration and profiles) and only the common fields are overwritten by the ``YML`` file.::
```
Comment (bentsherman) [inline docs/cli.rst:1269]: The params file can be YAML or JSON. I would say YAML/JSON file instead of `YML` file.

Comment (bentsherman) [reply docs/cli.rst:1269]: Also @pditommaso we should clarify what happens if you provide params through a config file and a params file via CLI arguments. I think the params file takes priority but I'm not certain.

Comment (abhi18av) [reply docs/cli.rst:1269]: Yes, this question has come up recently https://github.com/nextflow-io/nextflow/issues/2662

File: docs/cli.rst [hunk 4]
```diff
@@ -1265,6 +1265,12 @@ facilitates rapid iterations, inspections of any pipeline as well as debugging.
     $ nextflow run nextflow-io/hello -with-tower
  
 
+- Invoke the nextflow pipeline execution with a custom parameters ``YML`` file.
```
Comment (abhi18av) [inline docs/cli.rst:1268]:
```suggestion
- Invoke the nextflow pipeline execution with a custom parameters ``YAML/JSON`` file. 
```

File: docs/cli.rst (hunk 3 above)
Comment (abhi18av) [inline docs/cli.rst:1269]:
```suggestion
The parameters which are specified through this mechanism are merged with the resolved configuration (base configuration and profiles) and only the common fields are overwritten by the ``YAML/JSON`` file.::
```

**Review by abhi18av — COMMENTED** [review]
Accomodate Ben's review about the JSON optoin.

File: docs/cli.rst (hunk 2 above)
Comment (pditommaso) [reply docs/cli.rst:1294]: Actually I was wrong here. `self-update` and `-self-update` update both nextflow runtime and the launcher script. 


`NXF_VER` is meant to enforce the use of a specific nf version. Don't think it should be mentioned in this context.

**Review by pditommaso — APPROVED** [review]
Ok, updated it removing the refence to the NXF_VER. Eventually it could be added into a *tip* section

File: docs/cli.rst (hunk 3 above)
Comment (pditommaso) [reply docs/cli.rst]: Was missing this conversation. The CLI params overrides the ones the json/yaml file.

You can see it here, it reads the params file and put it into the `result` map. then if there are cli `params` copy them over the `result` map. 

https://github.com/nextflow-io/nextflow/blob/0f63f1a7cc7953ed1d014e8d3f821ba2941c6177/modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy#L496-L513

**Pull request merged**